### PR TITLE
Add support for new company attributes

### PR DESCRIPTION
**Pull request by thewheat**
Addresses https://github.com/intercom/intercom-java/issues/179 ~& https://github.com/intercom/intercom-java/issues/202~
- Add support for some missing company attributes
~- Add tests including~

### CODE REVIEW

File: intercom-java/src/main/java/io/intercom/api/Company.java [hunk 1]
```diff
@@ -258,15 +300,14 @@ public int getSessionCount() {
...
-     * session value for a company. Calling this method has no
-     * effect.
+     * Note. The Intercom API does not support changing the
+     * session value for a company. Method needed for testing
      *
-     * @param sessionCount this value is ignored
+     * @param sessionCount
      * @return the company object
      */
-    @Deprecated
```
Comment (thewheat) [inline intercom-java/src/main/java/io/intercom/api/Company.java]: Needed to remove this as it was preventing tests from passing. 
Open to alternative approaches to this

It is possible to remove `setSessionCount method` completely and Jackson will automatically read and extract the correct `sessionCount` value

File: intercom-java/src/main/java/io/intercom/api/Company.java [hunk 2]
```diff
@@ -257,19 +299,6 @@ public int getSessionCount() {
...
-     * session value for a company. Calling this method has no
-     * effect.
-     *
-     * @param sessionCount this value is ignored
-     * @return the company object
-     */
-    @Deprecated
-    public Company setSessionCount(int sessionCount) {
-        return this;
-    }
```
Comment (thewheat) [inline intercom-java/src/main/java/io/intercom/api/Company.java]: Removed this as it 
- prevented tests from passing 
- and doesn't seem to do anything of use and is probably overriding default behaviour causing and is causing issues such as https://github.com/intercom/intercom-java/issues/202 as well

File: intercom-java/src/main/java/io/intercom/api/Company.java [hunk 3]
```diff
@@ -353,6 +392,9 @@ public boolean equals(Object o) {
         if (untag != null ? !untag.equals(company.untag) : company.untag != null) return false;
         //noinspection RedundantIfStatement
         if (userCount != null ? !userCount.equals(company.userCount) : company.userCount != null) return false;
+        if (size != company.size) return false;
+        if (website != null ? !website.equals(company.website) : company.website != null) return false;
```
Comment (jonnyom) [inline intercom-java/src/main/java/io/intercom/api/Company.java]: This is quite complicated to read, you want to make sure that the current website and company website aren't equal, right? Is there any way you could export the equals checks to a separate method that returns the boolean?

Comment (thewheat) [reply intercom-java/src/main/java/io/intercom/api/Company.java]: Yeah I can see that it is essentially checking for the possibly inequality between the 2 objects. I was following the standard format that exists for string checking and it is utilised in several places. Could definitely use a refactor though 👍 

Do you see any issues with `if (!Objects.equals(website, company.website)) return false;` https://docs.oracle.com/javase/7/docs/api/java/util/Objects.html#equals(java.lang.Object,%20java.lang.Object)

I can try implement that as it seems built but seems to require Java 1.7

Comment (jonnyom) [reply intercom-java/src/main/java/io/intercom/api/Company.java]: That's definitely a lot easier to read. What version of Java is this on? Why isn't it up higher?

That would definitely be nice, but I can understand that you're following the format. As well though, maintainability and readability are definitely important 😄 

This definitely isn't a breaking change though.

File: intercom-java/src/test/java/io/intercom/api/CompanyTest.java [hunk 4]
```diff
@@ -0,0 +1,222 @@
+package io.intercom.api;
```
Comment (jonnyom) [inline intercom-java/src/test/java/io/intercom/api/CompanyTest.java]: Delighted that you're adding these tests, but I'm not actually sure that they're running in CI? Taking a look I can't see any specific suites listed 🤔

Comment (jonnyom) [reply intercom-java/src/test/java/io/intercom/api/CompanyTest.java]: Oh maybe they are 🤔

File: intercom-java/src/main/java/io/intercom/api/Company.java (hunk 3 above)
Comment (thewheat) [reply intercom-java/src/main/java/io/intercom/api/Company.java]: Don't think we have specified a version of Java we support but created this issue https://github.com/intercom/intercom-java/issues/214 to track the possibility of simplifying this

Comment (jonnyom) [reply intercom-java/src/main/java/io/intercom/api/Company.java]: Sorry Tim, I missed this. I think you're definitely right we need to look into simplifying it.

File: intercom-java/src/main/java/io/intercom/api/Company.java [hunk 5]
```diff
@@ -262,6 +304,15 @@ public Company setRemoteCreatedAt(long remoteCreatedAt) {
         return this;
     }
 
+    public long getLastRequestAt() {
```
Comment (jonnyom) [inline intercom-java/src/main/java/io/intercom/api/Company.java]: Should we set the entities lastRequestAt as well? As we do with the other API attributes?

Comment (thewheat) [reply intercom-java/src/main/java/io/intercom/api/Company.java]: @jonnyom not sure I get that 😅 Isn't this the `setLastRequestAt` below this what you're referring to? Or maybe I'm missing something

**Comment by SeanHealy33** [comment]
👍 nice work @thewheat

**Pull request merged**